Source/WebCore/dom/ContentVisibilityDocumentState.cpp

```diff
@@ -163,21 +193,28 @@ void ContentVisibilityDocumentState::updateContentRelevancyForScrollIfNeeded(con
         for (auto target : m_observer->observationTargets()) {
             if (target) {
                 ASSERT(target->renderer() && target->renderStyle()->contentVisibility() == ContentVisibility::Auto);
-                updateOnScreenObservationTarget(*target, false);
+                updateViewportProximity(*target, ViewportProximity::Far);
             }
         }
-        updateOnScreenObservationTarget(*scrollAnchorRoot, true);
-        scrollAnchorRoot->document().scheduleContentRelevancyUpdate(ContentRelevancy::OnScreen);
+        updateViewportProximity(*scrollAnchorRoot, ViewportProximity::Near);
         scrollAnchorRoot->document().updateRelevancyOfContentVisibilityElements();
     }
 }
 
-void ContentVisibilityDocumentState::updateOnScreenObservationTarget(const Element& element, bool onScreen)
+void ContentVisibilityDocumentState::updateViewportProximity(const Element& element, ViewportProximity viewportProximity)
+{
+    // No need to schedule content relevancy update for first time call, since
+    // that will be handled by determineInitialVisibleContentVisibility.
+    if (m_viewportProximity.contains(element))
```

Comment:
It's confusing that you have an argument which is an enum type called `viewportProximity` and a member variable called `m_viewportProximity` which is a collection of elements. I suggest you rename `m_viewportProximity`.